buildtools/pipenv/pipenv_test.go

```diff
@@ -0,0 +1,129 @@
+package pipenv_test
+
+import (
+	"errors"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/fossas/fossa-cli/buildtools/pipenv"
+	"github.com/fossas/fossa-cli/pkg"
+)
+
+type MockExec struct {
+	File  string
+	Error error
+}
+
+func (m MockExec) DataAccess(s string) (string, error) {
+	return m.File, m.Error
+}
+
+func newMockPipenv(file []byte, err error) pipenv.SystemPipenv {
+	return pipenv.SystemPipenv{
+		PipenvExec: MockExec{
+			File:  string(file),
+			Error: err,
+		},
+	}
+}
+
+/*
+	├── a@1.0.0
+	└─┬ b@2.0.0
+      ├─┬ c@3.0.0
+      │	└── d@4.0.0
+      └── e@5.0.0
+*/
+func TestDirectDeps(t *testing.T) {
+	file, err := ioutil.ReadFile("testdata/get-deps/pipenv-graph-json-tree.json")
+	assert.NoError(t, err)
+	testEnv := newMockPipenv(file, nil)
+
+	deps, error := testEnv.Deps()
+	assert.NoError(t, error)
+
+	imports := deps.Direct
+	assert.NotZero(t, imports)
+	assert.Equal(t, 2, len(imports))
+	assertImport(t, imports, "a", "1.0.0")
+	assertImport(t, imports, "b", "2.0.0")
+
+}
+
+func TestTransitiveDeps(t *testing.T) {
+	file, err := ioutil.ReadFile("testdata/get-deps/pipenv-graph-json-tree.json")
+	assert.NoError(t, err)
+	testEnv := newMockPipenv(file, nil)
+
+	deps, error := testEnv.Deps()
+	assert.NoError(t, error)
+
+	graph := deps.Transitive
+	assert.NotZero(t, graph)
+
+	packageA := findPackage(graph, "a", "1.0.0")
+	assert.NotZero(t, packageA)
+	assert.Equal(t, 1, len(packageA.Imports))
+	assertImport(t, packageA.Imports, "e", "5.0.0")
+
+	packageB := findPackage(graph, "b", "2.0.0")
+	assert.NotZero(t, packageB)
+	assert.Equal(t, 2, len(packageB.Imports))
+	assertImport(t, packageB.Imports, "c", "3.0.0")
+	assertImport(t, packageB.Imports, "e", "5.0.0")
+
+	packageC := findPackage(graph, "c", "3.0.0")
+	assert.NotZero(t, packageC)
+	assert.Equal(t, 1, len(packageC.Imports))
+	assertImport(t, packageC.Imports, "d", "4.0.0")
+
+	packageD := findPackage(graph, "d", "4.0.0")
+	assert.NotZero(t, packageD)
+	assert.Equal(t, 0, len(packageD.Imports))
+
+	packageE := findPackage(graph, "e", "5.0.0")
+	assert.NotZero(t, packageE)
+	assert.Equal(t, 0, len(packageE.Imports))
+}
+
+func TestNoFile(t *testing.T) {
+	testEnv := newMockPipenv([]byte{}, errors.New("test error"))
+
+	deps, err := testEnv.Deps()
+	assert.Zero(t, deps)
+	assert.EqualError(t, err, "test error")
+}
+
+func TestBadFile(t *testing.T) {
+	file, err := ioutil.ReadFile("testdata/get-deps/bad.json")
+	assert.NoError(t, err)
+
+	testEnv := newMockPipenv(file, nil)
+
+	deps, err := testEnv.Deps()
+	assert.Zero(t, deps)
+	assert.EqualError(t, err, "could not unmarshall JSON into dependency list: json: cannot unmarshal string into Go value of type []pipenv.dependency")
```

Comment:
I'm not sure you need to test this, since this is really a test of `json.Unmarshal`. The only exception is if you plan on returning a _specific_ error in the case of bad files. That might be useful for providing the user with useful debug information, but is irrelevant in this case because we're loading a test fixture into a mock tool, and the goodness of this file is irrelevant.

Reply:
I agree with @liftM here. I think it's reasonable to assume that `pipenv`'s output will be json consistently.